### PR TITLE
Split lib.rs into modules, and few minor improvements

### PR DESCRIPTION
**Pull request by poma**
* Accept filenames as command line arguments for Powers of Tau similar to Phase2
* Removed bn12_381 (looks like it conflicts with current code base anyways), and small_bn256 (if custom size is needed it's better to introduce a parameter for it instead of making a separate mod)
* add `test.sh` to phase2 similar to powers of tau

### CODE REVIEW

**Comment by poma** [comment]
there is a `test.sh` in phase2 dir now, but looks like I forgot to include patch_vk.js and cli.js that it uses in this PR (I think I need to move them to more appropriate location first, like under utils or something)

**Comment by poma** [comment]
Oh, didn't notice that I did a PR from master, so now it also contains refactorings that split lib.rs into modules

**Comment by kobigurk** [comment]
I tested most of it and it looks great!
`test.sh` in phase2 doesn't work because we're missing an example circuit.circom, patch_vk.js and cli.js

**Comment by poma** [comment]
I wanted to make example files prettier before committing them, in my repo they just hang around in `/phase2`, which is not an appropriate place for them

**Comment by kobigurk** [comment]
> I wanted to make example files prettier before committing them, in my repo they just hang around in `/phase2`, which is not an appropriate place for them

Sounds good, it would be better for them to be in a separate directory

File: phase2/src/bin/prove.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        &proof,
+        &input[1..]
+    ).unwrap();
+    assert!(result, "Proof is correct");
+
+    let repr_to_big = |r| {
+        BigUint::from_str_radix(&format!("{}", r)[2..], 16).unwrap().to_str_radix(10)
+    };
+    let repr_to_big2 = |r| {
+        BigUint::from_str_radix(&format!("{}", r)[2..], 16).unwrap().to_str_radix(10)
```
Comment (rawfalafel) [inline phase2/src/bin/prove.rs]: Should `repr_to_big` and `repr_to_big2` look identical here. Is that intentional?

Comment (poma) [reply phase2/src/bin/prove.rs]: Yes, but probably there is a better way to do this. Lambdas resolve their type arguments when first used, so I can't apply a single one to both `Fr` and `Fq` types.

Also this conversion code is duplicated in `export_keys` and needs some refactoring

Comment (kobigurk) [reply phase2/src/bin/prove.rs]: Those are good points. If we extract that into a proper function, we can use traits instead. Are you happy to do that @poma? If you give me commit access to the branch I'm happy to help as well.

**Comment by poma** [comment]
I don't know rust well enough to do that cleanly so I've added you as a collaborator to my repo

File: phase2/src/bin/prove.rs (hunk 1 above)
Comment (poma) [reply phase2/src/bin/prove.rs]: https://github.com/kobigurk/phase2-bn254/pull/4#issuecomment-577653412

Comment (poma) [reply phase2/src/bin/prove.rs]: I don't know rust well enough to do that cleanly so I've added you as a collaborator to my repo

**Pull request merged**